### PR TITLE
CloudCasa Helm Chart

### PR DESCRIPTION
**Pull request by bobadair**
Hi. This is the first submission of our CloudCasa agent Helm chart.
See https://cloudcasa.io/ for additional info on the service.

### CODE REVIEW

File: charts/cloudcasa/v0.1.0/questions.yml
Comment (pennyscissors) [inline charts/cloudcasa/v0.1.0/questions.yml]: Please remove these, they are not necessary. This repository is for Helm 3 charts only, and we only support Helm 3 in Rancher v2.4+ so your chart will be made available in all Rancher releases that support this catalog.

Comment (bobadair) [reply charts/cloudcasa/v0.1.0/questions.yml]: Thanks @PennyScissors. We'll make that change.

**Comment by bobadair** [comment]
Done. Please let us know if there are any other changes we should make, @PennyScissors. Thanks again.

**Comment by PennyScissors** [comment]
@bobadair Has this chart been tested in Rancher? What steps did you take to install it? I tried installing it, but it failed. The chart creates its own namespace which will cause conflict because Rancher creates a namespace for it.

File: charts/cloudcasa/v0.1.0/templates/cluster-register.yaml
Comment (pennyscissors) [inline charts/cloudcasa/v0.1.0/templates/cluster-register.yaml]: Rancher relies on Helm's built-in objects so that it can install the chart in its own namespace, or change the name, etc. So hardcoded namespaces and names like this could cause conflicts.
```yaml
  name: {{ .Release.Name }}
  namespace: {{ .Release.Namespace }}
```

**Comment by pennyscissors** [comment]
Hello, @bobadair. I tried installing the chart, but it failed. The chart creates its own namespace which will cause conflicts because Rancher creates a namespace for it. Please use Rancher to validate the chart installation, thanks
image

**Comment by nitishdsharma** [comment]
> Hello, @bobadair. I tried installing the chart, but it failed. The chart creates its own namespace which will cause conflicts because Rancher creates a namespace for it. Please use Rancher to validate the chart installation, thanks
> image

@PennyScissors 
Thanks for the suggestion on using the helm built-in objects. I have added it in Release Name in the template manifest. But our namespace is tightly coupled and we want it to be created from the chart itself. Request you to please answer the below query.

**Query:** How does Rancher create the cloudcasa-io namespace, looks like user type the namespace cloudcasa-io on Rancher UI while installing the chart? 

We actually install the Cloudcasa Rancher Apps Chart in the default namespace and chart itself creates the required namespace (cloudcasa-io) and other cloudcasa objects (Deployment, pods and service account). Please share your comments.

Thanks.

**Comment by bobadair** [comment]
Hi @PennyScissors. @nitishdsharma pushed a few commits that we think should resolve this. Can you take a look?

**Comment by pennyscissors** [comment]
Hello @bobadair @nitishdsharma, I re-tested the chart in Rancher v2.5.5 and it installed successfully 🎉 

The last step before I can approve and merge it is to use Helm's chart-testing tool to lint the chart. This is part of our CI, but it is not working properly at the moment. Could you please use their tool directly from their repo [here](https://github.com/helm/chart-testing)? I ran it on your chart locally and it failed. As for the maintainers failure you will encounter when using the tool, you can set the config to skip it.

If you haven't used that tool before, you will need to add/fetch the rancher/helm3-charts remote to your fork:
```
git remote add -f -t master helm3-charts https://github.com/rancher/helm3-charts.git
```

And your ct config should look similar to this (Might need some tweaking due to our chart dir structure):
```
remote: helm3-charts
target-branch: master
chart-dirs:
  - charts/cloudcasa
validate-maintainers: false
```

Output:
```
Linting charts...
Directory 'charts/cloudcasa' is not a valid chart directory. Skipping...
------------------------------------------------------------------------------------------------------------------------
 Charts to be processed:
------------------------------------------------------------------------------------------------------------------------
 cloudcasa => (version: "0.1.0", path: "charts/cloudcasa/v0.1.0")
------------------------------------------------------------------------------------------------------------------------

Linting chart 'cloudcasa => (version: "0.1.0", path: "charts/cloudcasa/v0.1.0")'
Checking chart 'cloudcasa => (version: "0.1.0", path: "charts/cloudcasa/v0.1.0")' for a version bump...
Unable to find chart on master. New chart detected.
Validating /cloudcasa-helm3-charts/charts/cloudcasa/v0.1.0/Chart.yaml...
Validation success! 👍
charts/cloudcasa/v0.1.0/values.yaml
  49:57     error    trailing spaces  (trailing-spaces)
  52:53     error    trailing spaces  (trailing-spaces)

------------------------------------------------------------------------------------------------------------------------
 ✖︎ cloudcasa => (version: "0.1.0", path: "charts/cloudcasa/v0.1.0") > Error waiting for process: exit status 1
------------------------------------------------------------------------------------------------------------------------
Error: Error linting charts: Error processing charts
Error linting charts: Error processing charts
```

**Comment by bobadair** [comment]
Thanks @PennyScissors. I will run ```ct lint``` on it next week. In the meantime, I've pushed a commit that removes trailing spaces from the lines that it complained about.

**Comment by bobadair** [comment]
Hi @PennyScissors. I believe ```ct lint``` should run cleanly now.

**Review by pennyscissors — APPROVED** [review]

**Pull request merged**
